### PR TITLE
[command] single user mode - use config object & add tests

### PR DESCRIPTION
**Pull request by MorrisJobke**
I migrated this command from `OC_Config` to `OC\SystemConfig` and added tests ;)

cc @deepdiver1975 @rullzer @Xenopathic

### CODE REVIEW

**Comment by Xenopathic** [comment]
:+1:

**Comment by th3fallen** [comment]
:+1: waiting on tests to finish but should be good.

File: core/command/maintenance/singleuser.php [hunk 1]
```diff
@@ -27,8 +27,21 @@
...
+
 class SingleUser extends Command {
 
+	/** @var SystemConfig */
+	protected $systemConfig;
+
+	/**
+	 * @param SystemConfig $config
+	 */
+	public function __construct(SystemConfig $systemConfig) {
```
Comment (DeepDiver1975) [inline core/command/maintenance/singleuser.php]: IConfig ??? to use public api only?

Comment (MorrisJobke) [inline core/command/maintenance/singleuser.php]: I just thought that it also needs to be usable before installation. But I noticed that this is not the case.

**Comment by MorrisJobke** [comment]
@DeepDiver1975 I fixed it :)

**Pull request merged**